src/components/dialogs/generator-modification-dialog.js

```diff
@@ -153,31 +160,68 @@ const GeneratorModificationDialog = ({
             : { id: '' };
     }, [formValues]);
 
-    const [generatorInfos, generatorIdField] = useAutocompleteField({
+    const [generatorId, generatorIdField] = useAutocompleteField({
         label: 'ID',
         validation: { isFieldRequired: true },
         inputForm: inputForm,
         formProps: filledTextField,
-        values: equipmentOptions?.sort(compareById),
+        values: equipmentOptions?.sort((a, b) => a.localeCompare(b)),
         allowNewValue: true,
         getLabel: getIdOrSelf,
         defaultValue:
-            equipmentOptions?.find((e) => e.id === formValueEquipmentId?.id) ||
+            equipmentOptions?.find((e) => e === formValueEquipmentId?.id) ||
             formValueEquipmentId,
         loading: loadingEquipmentOptions,
     });
 
+    const id = useMemo(() => {
```

Comment:
this method is a bit confusing, too many ways to get ids, it might be improved after dialog refactoring

Reply:
It's temporary, I added this method to avoid breaking the useAutocompleteField component in modification forms. it will be improved with the refactoring.